options/rtdl/generic/linker.cpp

```diff
@@ -1293,9 +1310,19 @@ void Loader::_buildTlsMaps() {
 	}
 }
 
-void Loader::initObjects() {
+void Loader::initObjects(SharedObject *mainExecutable) {
```

Comment:
Passing the main executable to this function is a bit ad-hoc. Can't we just store it in the `Loader` class instead?